client/src/sections/users/usersUpdateDelete.tsx

```diff
@@ -0,0 +1,240 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+/* eslint-disable @typescript-eslint/strict-boolean-expressions */
+/* eslint-disable @typescript-eslint/no-confusing-void-expression */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import React, { useState, Fragment } from 'react';
+import API from '@/src/apis';
+import { useRouter } from 'next/router';
+import { useForm } from 'react-hook-form';
+import RFInputField from '@/src/shared/components/ReactForm/RFInputField';
+import RFSelectField from '@/src/shared/components/ReactForm/RFSelectField';
+import Modal from '@/src/shared/components/Modal/Modal';
+import XmarkIcon from '@/src/shared/icons/XmarkIcon';
+import EditIcon from '@/src/shared/icons/EditIcon';
+
+interface UserUpdateDeleteProps {
+  id: number;
+}
+
+export interface UserData {
+  id: number;
+  username: string;
+  role_id: string;
+  email: string;
+  first_name: string;
+  last_name: string;
+  password: string;
+  confirm_password: string;
+}
+
+export const roles = [
+  { value: '1', label: 'Admin' },
+  { value: '3', label: 'Trainer' },
+  { value: '4', label: 'Trainee' }
+];
+
+export const initialUserData = {
+  id: 0,
+  username: '',
+  role_id: '',
+  email: '',
+  first_name: '',
+  last_name: '',
+  password: '',
+  confirm_password: ''
+};
+
+const UserEditDelete: React.FC<UserUpdateDeleteProps> = ({ id }) => {
+  const router = useRouter();
+  const params = router.query;
+
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+  const [, setUserData] = useState<UserData>(initialUserData);
+
+  const {
+    register,
+    reset,
+    handleSubmit,
+    watch,
+    formState: { errors }
+  } = useForm<UserData>();
+
+  const onSubmit = async (data: UserData) => {
+    const password = watch('password');
+    const confirmPassword = watch('confirm_password');
+    if (password !== confirmPassword) {
+      alert('Passwords do not match');
+      return;
+    }
+    const confirmed = window.confirm(
+      'Are you sure you want to update the user data?'
+    );
+    if (!confirmed) {
+      return;
+    }
+    try {
+      await API.put(`user/${params.company_id}/${id}`, data);
+      alert('User data updated successfully!');
```

Comment:
instead of using the native alert, you can just use the toastify package (it is already implemented) .  Chech the utils folder.
- apply to all